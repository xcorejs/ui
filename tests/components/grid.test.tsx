import 'jest-styled-components';

import { Box, Grid, LoremIpsum } from '@xcorejs/ui';
import React from 'react';
import renderer from 'react-test-renderer';

test('Grid component', () => {
  const component = renderer.create(
    <>
      <Grid columns="repeat(3, 75px) auto" rows="200px auto 70px" gap="10px" p="10px" height="100%">
        <Box column="1 / span 3" row="1" placeSelf="center">Logo</Box>
        <Box column="1 / span 3" row="2">
          Menu
        </Box>
        <Box column="1 / span 2" row="3" alignSelf="center">
          User bar
        </Box>
        <Box column="3 / span 1" row="3" placeSelf="center">
          <img
            src="https://avatars0.githubusercontent.com/u/15893340?s=460&v=4"
            alt="avatar"
            style={{ width: '64px', height: '64px' }}
          />
        </Box>

        {/* Content */}
        <Box column="4 / span 1" row="1" placeSelf="center" >Header</Box>
        <Box column="4 / span 1" row="2">
          <Box>
            <LoremIpsum count={2} />
          </Box>
          <Box>
            <LoremIpsum count={2} />
          </Box>
        </Box>

        {/* Footer */}
        <Box column="4 / span 1" row="3" placeSelf="center">
          Footer
        </Box>
      </Grid>
    </>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
