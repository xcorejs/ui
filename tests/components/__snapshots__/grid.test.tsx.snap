// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid component 1`] = `
.c1 {
  grid-column: 1 / span 3;
  grid-row: 1;
  place-self: center;
}

.c2 {
  grid-column: 1 / span 3;
  grid-row: 2;
}

.c3 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  grid-column: 1 / span 2;
  grid-row: 3;
}

.c4 {
  grid-column: 3 / span 1;
  grid-row: 3;
  place-self: center;
}

.c5 {
  grid-column: 4 / span 1;
  grid-row: 1;
  place-self: center;
}

.c6 {
  grid-column: 4 / span 1;
  grid-row: 2;
}

.c8 {
  grid-column: 4 / span 1;
  grid-row: 3;
  place-self: center;
}

.c7 {
  font-size: 1.5rem;
  font-family: "rubik",-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  font-size: 1.5rem;
  line-height: 2rem;
  -webkit-transition: color 300ms;
  transition: color 300ms;
  color: #1E3441;
}

.c0 {
  height: 100%;
  padding: 10px;
  display: grid;
  display: -ms-grid;
  grid-template-columns: repeat(3,75px) auto;
  grid-gap: 10px;
  -ms-grid-columns: 75px 10px 75px 10px 75px 10px auto;
  -ms-grid-rows: 200px 10px auto 10px 70px;
}

.c0 > *:nth-child(1) {
  -ms-grid-column: 1;
  -ms-grid-row: 1;
}

.c0 > *:nth-child(2) {
  -ms-grid-column: 3;
  -ms-grid-row: 1;
}

.c0 > *:nth-child(3) {
  -ms-grid-column: 5;
  -ms-grid-row: 1;
}

.c0 > *:nth-child(4) {
  -ms-grid-column: 7;
  -ms-grid-row: 1;
}

.c0 > *:nth-child(5) {
  -ms-grid-column: 1;
  -ms-grid-row: 3;
}

.c0 > *:nth-child(6) {
  -ms-grid-column: 3;
  -ms-grid-row: 3;
}

.c0 > *:nth-child(7) {
  -ms-grid-column: 5;
  -ms-grid-row: 3;
}

.c0 > *:nth-child(8) {
  -ms-grid-column: 7;
  -ms-grid-row: 3;
}

.c0 > *:nth-child(9) {
  -ms-grid-column: 1;
  -ms-grid-row: 5;
}

.c0 > *:nth-child(10) {
  -ms-grid-column: 3;
  -ms-grid-row: 5;
}

.c0 > *:nth-child(11) {
  -ms-grid-column: 5;
  -ms-grid-row: 5;
}

.c0 > *:nth-child(12) {
  -ms-grid-column: 7;
  -ms-grid-row: 5;
}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:85em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

<div
  className="c0"
  height="100%"
  rows={
    Array [
      "200px auto 70px",
      null,
      null,
      null,
      null,
      null,
    ]
  }
>
  <div
    className="c1"
  >
    Logo
  </div>
  <div
    className="c2"
  >
    Menu
  </div>
  <div
    className="c3"
  >
    User bar
  </div>
  <div
    className="c4"
  >
    <img
      alt="avatar"
      src="https://avatars0.githubusercontent.com/u/15893340?s=460&v=4"
      style={
        Object {
          "height": "64px",
          "width": "64px",
        }
      }
    />
  </div>
  <div
    className="c5"
  >
    Header
  </div>
  <div
    className="c6"
  >
    <div
      className=""
    >
      <span
        className="c7"
        color="text"
        fontFamily="text"
        fontSize="1.5rem"
      >
        Ad ad ad ad ad. Ad ad ad ad ad. Ad ad ad ad ad.
Ad ad ad ad ad. Ad ad ad ad ad. Ad ad ad ad ad.
      </span>
    </div>
    <div
      className=""
    >
      <span
        className="c7"
        color="text"
        fontFamily="text"
        fontSize="1.5rem"
      >
        Ad ad ad ad ad. Ad ad ad ad ad. Ad ad ad ad ad.
Ad ad ad ad ad. Ad ad ad ad ad. Ad ad ad ad ad.
      </span>
    </div>
  </div>
  <div
    className="c8"
  >
    Footer
  </div>
</div>
`;
